### ISSUE TITLE
Need tester ;)

### ISSUE THREAD

**Issue by mch1307** [mch1307/gomotics#7]
@abbar21 I am ready to release a new version, the main new functionality is that it can automatically create objects and scripts in Jeedom. Would you be interesting in testing it?
thanks

**Comment by abbar21** [comment]
Yes sure ! I'm busy this evening but it should be possible tomorrow.

**Comment by webalexeu** [comment]
I can perform test also if you want

**Comment by mch1307** [comment]
Thank you!
Release v0.4.0 is available. It will only work with Jeedom v3.1.8+, which is beta for now.
So I advise you to test against a Jeedom test system, in beta mode

**Comment by abbar21** [comment]
Hi,
I don't forget you but today i've 404 error with https://github.com/jeedom/core/archive/beta.zip so I'm not able to update my new VM testJeedom.

**Comment by mch1307** [comment]
Hi, just install a new jeedom instance and follow [this doc](https://github.com/jeedom/documentation/blob/master/howto/fr_FR/passer_en_beta.asciidoc) to put it in beta.
You also need to install the script plugin from github

**Comment by abbar21** [comment]
I know how to update jeedom but the URL to update to beta is not responding.
Here is a part of the update log from jeedom
017-11-17 10:59:30--  https://codeload.github.com/jeedom/core/zip/beta
Resolving codeload.github.com (codeload.github.com)... 192.30.253.121, 192.30.253.120
Connecting to codeload.github.com (codeload.github.com)|192.30.253.121|:443... connected.
HTTP request sent, awaiting response... 404 Not Found
2017-11-17 10:59:30 ERROR 404: Not Found.

Why update script plugin from GitHub and not from market ?

**Comment by mch1307** [comment]
works for me...
Just configure as described in the doc, should be ok
The script plugin has also been enhanced, I have installed it from github but I guess you can choose the beta version from the market

**Comment by abbar21** [comment]
Always the same problem... despite the doc.
I leaved a message in the jeedom forum.
https://www.jeedom.com/forum/viewtopic.php?f=23&t=31619

**Comment by abbar21** [comment]
It works ! But in the conf.d (supervisor) I did not specify AUTO_CREATE_OBJECTS to True but the objects were still created.
what's happening if I delete or update an object in NHC ? It's deleted or update automaticaly in Jeedom ?

**Comment by mch1307** [comment]
It will check the auto_Create_objects flag.
Currently there is no update/delete possible, and I don't think I will implement those.
The goal of the auto_create_objects is to ease the setup in Jeedom as it can be very long and error prone

**Comment by abbar21** [comment]
In the doc, the default flag of auto_Create_objects flag is false.

**Comment by mch1307** [comment]
thanks for your feedback.
Should be fixed in v0.4.1

**Comment by abbar21** [comment]
The objects were created, but the light does not work ...
[gomotics.log](https://github.com/mch1307/gomotics/files/1489186/gomotics.log)

**Comment by mch1307** [comment]
The default value for GM_HOSTPORT does not include the port, will fix this. This means the HTTP commands created in Jeedom are wrong. 
Can you delete all objects from Jeedom, set the GM_HOSTPORT variable to localhost:8081, clean the log and start gomotics. Set the AUTO_CREATE_OBJECTS=Y to re-create all objects in Jeedom

**Comment by abbar21** [comment]
I'm testing the 0.4.2 version with the last conf.d but not lights.
[gomotics.log](https://github.com/mch1307/gomotics/files/1493045/gomotics.log)
For Dimmer Salon
<img width="1645" alt="capture d ecran 2017-11-21 a 22 43 28" src="https://user-images.githubusercontent.com/31033427/33098427-b4713818-cf0d-11e7-94b7-c41fa5a81f7e.png">

**Comment by mch1307** [comment]
What do you exactly mean by no light?
When switching from NHC, is the state updated in Jeedom?
If switching from Jeedom, is the light physically switched?
Can you send one example of script setup in Jeedom, specifically the URL of the on or off command. What happens if you try this url from a browser?

**Comment by mch1307** [comment]
Assuming gomotics runs on the same host thans Jeedom, what happens if you type http://jeedom_host:8081/api/v1/jeedom/58/100 ?

**Comment by abbar21** [comment]
When I switch a light in NHC the state is returned in Jeedom.
But when if I switch a light in Jeedom, the light is not physically switched in NHC
<img width="558" alt="capture d ecran 2017-11-21 a 23 27 25" src="https://user-images.githubusercontent.com/31033427/33100310-30c4bf10-cf14-11e7-8d31-85aab6d52eca.png">

**Comment by mch1307** [comment]
can u try this http://192.168.1.38:8081/api/v1/nhc/58, it should return the "Dimmer Salon" info
58 is the NHC id, so I was wrong. Can u try to replace 58 by the "Dimmer Salon" id in Jeedom and retry http://192.168.1.38:8081/api/v1/jeedom/__ID__/100?

**Comment by abbar21** [comment]
<img width="529" alt="capture d ecran 2017-11-21 a 23 38 37" src="https://user-images.githubusercontent.com/31033427/33100627-57319082-cf15-11e7-84e8-0af3f2fb321f.png">
http://192.168.1.38:8081/api/v1/nhc/29 hasn't a return

**Comment by mch1307** [comment]
Try jeedom/29/100 , Not nhc

**Comment by abbar21** [comment]
<img width="567" alt="capture d ecran 2017-11-21 a 23 51 33" src="https://user-images.githubusercontent.com/31033427/33101033-f69f49f6-cf16-11e7-8a89-753bb06ce2b2.png">
 ;)

**Comment by mch1307** [comment]
And the light was switched on?
Are your sure you are using the beta version of the plugin "script"?

**Comment by abbar21** [comment]
Yes the light was switched on. and with jeedom/29/0 switched Off
Yes script - beta version 2017-10-04 01:03:18

**Comment by mch1307** [comment]
It is not the right version. I have this:
![image](https://user-images.githubusercontent.com/17752394/33114769-a8006850-cf5e-11e7-9bae-5e3d8730b26b.png)

**Comment by abbar21** [comment]
I'don't know how to update the plugin directly from GitHub :(

**Comment by mch1307** [comment]
Uninstall the current one, then go to Plugins \ gestion de plugins \ ajouter depuis une autre source
![image](https://user-images.githubusercontent.com/17752394/33116291-8486576c-cf64-11e7-9193-80694a101929.png)

**Comment by abbar21** [comment]
I can't choose Github just Market and URL

**Comment by mch1307** [comment]
You have to activate it first in configuration \ Misa à jours:
![image](https://user-images.githubusercontent.com/17752394/33126763-0209ce78-cf86-11e7-89f7-66c8b6f5dcb8.png)

**Comment by abbar21** [comment]
Thank you !
Now, I have the same version as you of the + jeedom plugin script
But the light switch does not work in the jeedom
It was with the 0.4.2... I had not seen the new version

**Comment by abbar21** [comment]
It looks like everything is good! I'm waiting for Jeedom + plugin script no longer in beta to put on my production server

**Comment by mch1307** [comment]
Good news!

**Comment by mch1307** [comment]
@webalexeu  did you have the opportunity to do some tests?
thanks

**Comment by webalexeu** [comment]
Hello,


Sorry, I upgrade my controller to version 2 and apparently, port 8000 is not opened anymore

I didn't had the time to ask the support how to access API


Regards,

Alex

________________________________
De : Michael Champagne <notifications@github.com>
Envoyé : vendredi 1 décembre 2017 21:21
À : mch1307/gomotics
Cc : webalexeu; Mention
Objet : Re: [mch1307/gomotics] Need tester ;) (#7)


@webalexeu<https://nam01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fwebalexeu&data=02%7C01%7Calexandre.jardon%40webalex.eu%7Cb878999c79e74c0ad74e08d538f91b4b%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C636477564906829647&sdata=5BeReFE2YLMBXGrRdPnuVErfewVbPPnoCRHbBpqfr2s%3D&reserved=0> did you have the opportunity to do some tests?
thanks

**Comment by mch1307** [comment]
Hi,
Interesting... Can you run a port scan on your controller IP using tool like [nmap](https://nmap.org)
This would allow to see which ports are opened.
Did you upgrade only the controller or the complete installation?

**Comment by webalexeu** [comment]
Hello,


I upgrade everything.


Here is the nmap result:


PORT     STATE  SERVICE         VERSION
22/tcp   open   ssh             OpenSSH 7.5 (protocol 2.0)
|
80/tcp   open   http            lighttpd 1.4.33
| http-methods:
|_  Supported Methods: OPTIONS GET HEAD POST
|_http-server-header: lighttpd/1.4.33
|_http-title: Site doesn't have a title (text/html).
3580/tcp open   http            Mono-HTTPAPI 1.0
|_http-cors: GET POST PUT DELETE PATCH
|_http-favicon: Unknown favicon MD5: 3691F05AF9B03B858F9393B0AE579D6B
| http-methods:
|   Supported Methods:  OPTIONS
|_  Potentially risky methods:
|_http-server-header: Mono-HTTPAPI/1.0
|_http-title: 404
5060/tcp open   sip-proxy       Asterisk PBX 13.17.0
5061/tcp open   ssl/sip-tls?
8000/tcp closed http-alt
8001/tcp open   vcom-tunnel?
8080/tcp closed http-proxy
8081/tcp closed blackice-icecap

Alex
________________________________
De : Michael Champagne <notifications@github.com>
Envoyé : mardi 5 décembre 2017 15:37
À : mch1307/gomotics
Cc : webalexeu; Mention
Objet : Re: [mch1307/gomotics] Need tester ;) (#7)


Hi,
Interesting... Can you run a port scan on your controller IP using tool like nmap<https://nam01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fnmap.org&data=02%7C01%7C%7C96030a8eb2c34fe298e908d53bedbdc4%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C636480814626196779&sdata=gBDXOdqHSayBerBwXhTgQhp1fkQCMtEO4tR5aAdPtxo%3D&reserved=0>
This would allow to see which ports are opened.
Did you upgrade only the controller or the complete installation?

**Comment by mch1307** [comment]
I have seen that even the Niko app is a different one, and requires the your installation is registered by them.. Difficult for me to get further without having the hardware.

**Comment by mch1307** [comment]
Your NHC control unit reference is 550-00003 or 550-00002?

**Comment by webalexeu** [comment]
I've sent a mail to support to see if there is any API


I hope that they put that in place with new version, let's see what they will reply


Alex

________________________________
De : Michael Champagne <notifications@github.com>
Envoyé : mercredi 6 décembre 2017 21:36
À : mch1307/gomotics
Cc : webalexeu; Mention
Objet : Re: [mch1307/gomotics] Need tester ;) (#7)


I have seen that even the Niko app is a different one, and requires the your installation is registered by them.. Difficult for me to get further without having the hardware.

**Comment by abbar21** [comment]
Hi,

Any news from Niko ?
I do not have the hardware to test NHC 2 but depending the new features maybe i'll change.

**Comment by webalexeu** [comment]
Hello,


Here is the answer from niko:

Bonjour Alexandre,



No on à pas une api pour NHC2.



Si vous avez encore des questions, n'hésitez pas à nous contacter.



Cordialement


I try the different port reported open by NMAP but without success until now


I will maybe downgrade back to version 1


It's very impressive that they don't propose any api in 2018



________________________________
De : abbar21 <notifications@github.com>
Envoyé : mercredi 27 décembre 2017 22:38
À : mch1307/gomotics
Cc : webalexeu; Mention
Objet : Re: [mch1307/gomotics] Need tester ;) (#7)


Hi,

Any news from Niko ?
I do not have the hardware to test NHC 2 but depending the new features maybe i'll change.

**Comment by mch1307** [comment]
It seems they are more focused on their "installateurs" rather than end-users.
I did not see really interesting new features and not having any API in 2018 is a real shame

**Comment by webalexeu** [comment]
I totally agree with you


I'm really seriously thinking to come back to version 1.

________________________________
De : Michael Champagne <notifications@github.com>
Envoyé : lundi 8 janvier 2018 09:04
À : mch1307/gomotics
Cc : webalexeu; Mention
Objet : Re: [mch1307/gomotics] Need tester ;) (#7)


It seems they are more focused on their "installateurs" rather than end-users.
I did not see really interesting new features and not having any API in 2018 is a real shame

**Comment by webalexeu** [comment]
Hello @mch1307 ,

I've good news for V2. I've found on OpenHab2 that they are currently upgrading their addon to work with NHC V2 (https://github.com/openhab/openhab2-addons/pull/4226) 
In V2, there is a MQTT Broker Server running on the controller (Mosquitto).
I've tested MQTT part and indeed, we can control and get status.

Let me know if you are interested to have more details

Alex
